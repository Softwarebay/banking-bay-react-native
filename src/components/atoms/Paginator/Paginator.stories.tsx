import { number } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { ViewStory } from 'utils/storybook/ViewStory';

import { Paginator } from './Paginator';

storiesOf('Atoms/Paginator', module)
	.addDecorator((getStory) => <ViewStory center>{getStory()}</ViewStory>)
	.add('Current at index 0', () => (
		<Paginator
			currentIndex={number('currentIndex', 0, {
				range: true,
				min: 0,
				max: 2,
				step: 1,
			})}
			slides={[{ id: 1 }, { id: 2 }, { id: 3 }]}
		/>
	))
	.add('Current at index 1', () => (
		<Paginator
			currentIndex={number('currentIndex', 1, {
				range: true,
				min: 0,
				max: 2,
				step: 1,
			})}
			slides={[{ id: 1 }, { id: 2 }, { id: 3 }]}
		/>
	))
	.add('Current at index 2', () => (
		<Paginator
			currentIndex={number('currentIndex', 2, {
				range: true,
				min: 0,
				max: 2,
				step: 1,
			})}
			slides={[{ id: 1 }, { id: 2 }, { id: 3 }]}
		/>
	));
