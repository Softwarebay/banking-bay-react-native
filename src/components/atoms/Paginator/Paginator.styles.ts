import { StyleSheet } from 'react-native';

import { colors } from 'theme';

export const styles = StyleSheet.create({
	current: {
		opacity: 1,
	},
	dot: {
		borderColor: colors.mainDark,
		borderRadius: 100,
		borderWidth: 3,
		height: 8,
		marginHorizontal: 8,
		opacity: 0.2,
		width: 8,
	},
});
