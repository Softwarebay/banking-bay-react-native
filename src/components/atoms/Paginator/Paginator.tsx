import { View } from 'react-native';

import { Flex } from '../Flex/Flex';

import { styles } from './Paginator.styles';

export interface Props {
	slides: { id: number }[];
	currentIndex: number;
	testID?: string;
	activeColor?: string;
}

export const Paginator = ({
	slides,
	currentIndex,
	testID,
	activeColor,
}: Props) => {
	return (
		<Flex direction="row" testID={testID}>
			{slides.map((slide, index) => (
				<View
					key={slide.id}
					style={[
						styles.dot,
						index === currentIndex && styles.current,
						{ borderColor: activeColor },
					]}
				/>
			))}
		</Flex>
	);
};
