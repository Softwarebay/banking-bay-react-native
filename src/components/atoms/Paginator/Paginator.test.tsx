import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { Paginator } from './Paginator';

import type { Props } from './Paginator';

describe('Paginator', () => {
	const props: Omit<Props, 'currentIndex'> = {
		slides: [{ id: 1 }, { id: 2 }],
		activeColor: 'blue',
		testID: 'paginator-test-id',
	};

	it('should match snapshot - active index 0', () => {
		const tree = testRenderer
			.create(<Paginator {...props} currentIndex={0} />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - active index 1', () => {
		const tree = testRenderer
			.create(<Paginator {...props} currentIndex={1} />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should should apply highlight color based on props', () => {
		render(<Paginator {...props} currentIndex={0} />);

		const paginator = screen.getByTestId('paginator-test-id');

		paginator.children.forEach((dot) =>
			expect(dot).toHaveStyle({
				borderColor: 'blue',
			})
		);
	});

	it('should highlight first paginator dot when current index is 0', () => {
		render(<Paginator {...props} currentIndex={0} />);

		const paginator = screen.getByTestId('paginator-test-id');

		const [currentDot, nextDot] = paginator.children;

		expect(currentDot).toHaveStyle({
			opacity: 1,
		});
		expect(nextDot).toHaveStyle({
			opacity: 0.2,
		});
	});

	it('should highlight second paginator dot when current index is 1', () => {
		render(<Paginator {...props} currentIndex={1} />);

		const paginator = screen.getByTestId('paginator-test-id');

		const [prevDot, currentDot] = paginator.children;

		expect(currentDot).toHaveStyle({
			opacity: 1,
		});
		expect(prevDot).toHaveStyle({
			opacity: 0.2,
		});
	});
});
