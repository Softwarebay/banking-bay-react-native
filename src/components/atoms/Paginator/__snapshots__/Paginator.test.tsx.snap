// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator should match snapshot - active index 0 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
  testID="paginator-test-id"
>
  <View
    style={
      Array [
        Object {
          "borderColor": "#1B1D4D",
          "borderRadius": 100,
          "borderWidth": 3,
          "height": 8,
          "marginHorizontal": 8,
          "opacity": 0.2,
          "width": 8,
        },
        Object {
          "opacity": 1,
        },
        Object {
          "borderColor": "blue",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderColor": "#1B1D4D",
          "borderRadius": 100,
          "borderWidth": 3,
          "height": 8,
          "marginHorizontal": 8,
          "opacity": 0.2,
          "width": 8,
        },
        false,
        Object {
          "borderColor": "blue",
        },
      ]
    }
  />
</View>
`;

exports[`Paginator should match snapshot - active index 1 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "row",
      },
      undefined,
    ]
  }
  testID="paginator-test-id"
>
  <View
    style={
      Array [
        Object {
          "borderColor": "#1B1D4D",
          "borderRadius": 100,
          "borderWidth": 3,
          "height": 8,
          "marginHorizontal": 8,
          "opacity": 0.2,
          "width": 8,
        },
        false,
        Object {
          "borderColor": "blue",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "borderColor": "#1B1D4D",
          "borderRadius": 100,
          "borderWidth": 3,
          "height": 8,
          "marginHorizontal": 8,
          "opacity": 0.2,
          "width": 8,
        },
        Object {
          "opacity": 1,
        },
        Object {
          "borderColor": "blue",
        },
      ]
    }
  />
</View>
`;
